ExternalModule.php

```diff
@@ -69,4 +69,40 @@ function includeJs($path) {
     protected function setJsSettings($settings) {
         echo '<script>redcapEntity = ' . json_encode($settings) . ';</script>';
     }
+
+    /**
+     * @return \string[][]
+     * @throws \Exception
+     */
+    public function getProjectList() {
+
+        // get the project list that conforms to the user rights of the current user
+        $sql = "SELECT p.project_id, p.app_title
+					FROM redcap_projects p, redcap_user_rights u
+					WHERE p.project_id = u.project_id
+						AND u.username = '" . db_real_escape_string( USERID ) . "'";
+        $externalModule = new ExternalModules();
+        $query = $externalModule->getFrameworkInstance( $this->PREFIX )->createQuery();
+
+        $query->add( $sql );
+        $queryResults = $query->execute();
```

Comment:
```suggestion
        $queryResults = $this->query( $sql );
```
`createQuery` is only available in framework version 4; in framework version 1 (which Entity is on), the module still has access to the (much less robust) `query` function.  
This accomplishes the same query without instantiating `ExternalModules`.